android/src/main/java/com/dylanvann/fastimage/FastImageViewModule.java

```diff
@@ -45,12 +53,39 @@ public void run() {
                             //    - data:image/png;base64
                             .load(
                                     imageSource.isBase64Resource() ? imageSource.getSource() :
-                                    imageSource.isResource() ? imageSource.getUri() : imageSource.getGlideUrl()
+                                            imageSource.isResource() ? imageSource.getUri() : imageSource.getGlideUrl()
                             )
                             .apply(FastImageViewConverter.getOptions(source))
                             .preload();
                 }
             }
         });
     }
+
+    @ReactMethod
+    public void deleteImage(final String url, final Promise promise) {
+        final Activity activity = getCurrentActivity();
+        if (activity == null) return;
+        Glide
+                .with(activity.getApplicationContext())
+                .asFile()
+                .load(url)
+                .listener(new RequestListener<File>() {
+                    @Override
+                    public boolean onLoadFailed(@Nullable GlideException e, Object model, Target<File> target, boolean isFirstResource) {
+
+                        return false;
+                    }
```

Comment:
I don't know much of Java, but shouldn't you reject the Promise if the `onLoadFailed` is called?

Reply:
Oh, sorry, u right. I will update request, thx